src/diffusers/modeling_utils.py

```diff
@@ -82,12 +87,16 @@ def find_tensor_attributes(module: torch.nn.Module) -> List[Tuple[str, Tensor]]:
         return first_tuple[1].dtype
 
 
-def load_state_dict(checkpoint_file: Union[str, os.PathLike]):
+def load_state_dict(checkpoint_file: Union[str, os.PathLike], is_pytorch: bool):
     """
     Reads a PyTorch checkpoint file, returning properly formatted errors if they arise.
     """
     try:
-        return torch.load(checkpoint_file, map_location="cpu")
+        if is_pytorch:
+            return torch.load(checkpoint_file, map_location="cpu")
+        else:
+            return load_file(checkpoint_file, device="cpu")
```

Comment:
just for readability could we maybe do the following:
```suggestion
            return safetensors.torch.load_file(checkpoint_file, device="cpu")
```

then the reader directly knows the loading comes from safetensors?

Reply:
Sure.